cmake/external/mkldnn.cmake

```diff
@@ -31,9 +31,17 @@ IF(APPLE)
     return()
 ENDIF()
 
-MESSAGE(STATUS "Set ${MKLDNN_INSTALL_DIR}/lib to runtime path")
+# Introduce variables:
+# * CMAKE_INSTALL_LIBDIR
+INCLUDE(GNUInstallDirs)
+SET(LIBDIR "lib")
+if(CMAKE_INSTALL_LIBDIR MATCHES ".*lib64$")
+  SET(LIBDIR "lib64")
+endif()
+
+MESSAGE(STATUS "Set ${MKLDNN_INSTALL_DIR}/l${LIBDIR} to runtime path")
 SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}" "${MKLDNN_INSTALL_DIR}/lib")
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}" "${MKLDNN_INSTALL_DIR}/${LIBDIR}")
```

Comment:
For windows, why 
https://github.com/PaddlePaddle/Paddle/pull/15116/files#diff-66252024e9fb29cab9d57ee555592772R103 is not updated here?

Reply:
I Can not ensure that use ${CMAKE_INSTALL_LIBDIR} to replace ${LIBDIR}.